src/NuGet.Clients/PackageManagement.PowerShellCmdlets/PackageActionBaseCommand.cs

```diff
@@ -59,23 +59,25 @@ public PackageActionBaseCommand()
         protected virtual void Preprocess()
         {
             CheckSolutionState();
-            try
+
+            var result = ValidateSource(Source, validateSource: true);
+            if (result.Validity == SourceValidity.UnknownSource)
```

Comment:
The error message is different in this case, thus the different handling. However, the validation could certainly be split into a separate method so this case doesn't get double validation.